data/campaigns/Heir_To_The_Throne/scenarios/05b_Isle_of_the_Damned.cfg

```diff
@@ -427,7 +479,48 @@
         y=7,7,8,7,7,7,8
     [/move_unit_fake]
 
-    {PLACE_IMAGE "units/transport/galleon.png~RC(magenta>red)" 27 8}
+    {PLACE_IMAGE "units/transport/galleon.png~RC(magenta>red)~FL()" 27 8}
+
+    [unstore_unit]
+        variable=hidden_advisor
+        x,y=27,8
+    [/unstore_unit]
+
+    {CLEAR_VARIABLE hidden_advisor}
+
+    [lua]
+        code=<<
+            -- This is a bit of a hack. The issue is we can not hide the unit
+            -- with [hide_unit] because the normal [message] seems to unhide
+            -- it. So this skips a **LOT** of the work the normal [message]
+            -- does, because we know exactly what we need for here, and it
+            -- hides the unit we just got with [unstore_unit] **THEN** fires
+            -- the highlight and redraw. Since this is all a cobbled together
+            -- mess, we also need to be sure to [kill] the unit!
+
+            function wesnoth.wml_actions.message_from_hidden_unit(cfg)
+                local speaker = wesnoth.get_units(cfg)[1]
+                speaker.hidden = true
+                wesnoth.highlight_hex(speaker.x, speaker.y)
+                wesnoth.fire("redraw")
+                wesnoth.show_message_dialog({
+                    left_side=nil,
+                    title=speaker.name,
+                    message=cfg.message,
+                    portrait=speaker.portrait
+                })
+            end
+        >>
```

Comment:
You actually don't need the parentheses around a function call when its argument is a single table - ie, `wesnoth.show_message_dialog{...}` will work. (You also don't need them when the argument is a single string, which explains why `wesnoth.textdomain "blah"` works.)

I'm just pointing this out so that you know. You don't need to feel pressured to update the PR to change it.